atc/worker/artifact_source.go

```diff
@@ -111,6 +114,18 @@ func (source *cacheArtifactSource) ExistsOn(logger lager.Logger, worker Worker)
 	return worker.FindVolumeForTaskCache(logger, source.TeamID, source.JobID, source.StepName, source.Path)
 }
 
+type CloseWithError struct {
+	NoErringCloser interface {
+		Close()
+	}
+}
+
+// I know this is bad
+func (cwe CloseWithError) Close() error {
```

Comment:
the new zstd reader does not follow io.Closer { Close() error} but rather just has Close() with no error. I had to wrap it so that it can be passed to fileReadMultiCloser.closers[]

I'm open to suggestions on this.